fakestorage/response.go

```diff
@@ -18,30 +19,36 @@ type listResponse struct {
 	Prefixes []string      `json:"prefixes"`
 }
 
-func newListBucketsResponse(bucketNames []string) listResponse {
+func newListBucketsResponse(buckets []backend.Bucket) listResponse {
 	resp := listResponse{
 		Kind:  "storage#buckets",
-		Items: make([]interface{}, len(bucketNames)),
+		Items: make([]interface{}, len(buckets)),
 	}
-	sort.Strings(bucketNames)
-	for i, name := range bucketNames {
-		resp.Items[i] = newBucketResponse(name)
+	for i, bucket := range buckets {
+		resp.Items[i] = newBucketResponse(bucket.Name, bucket.VersioningEnabled)
 	}
 	return resp
 }
 
 type bucketResponse struct {
-	Kind string `json:"kind"`
-	ID   string `json:"id"`
-	Name string `json:"name"`
+	Kind       string            `json:"kind"`
+	ID         string            `json:"id"`
+	Name       string            `json:"name"`
+	Versioning *bucketVersioning `json:"versioning,omitempty"`
 }
 
-func newBucketResponse(bucketName string) bucketResponse {
-	return bucketResponse{
-		Kind: "storage#bucket",
-		ID:   bucketName,
-		Name: bucketName,
+type bucketVersioning struct {
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+func newBucketResponse(bucketName string, versioningEnabled bool) (br bucketResponse) {
+	br = bucketResponse{
```

Comment:
Let's avoid the naked return.

```suggestion
	return bucketResponse{
```